### ISSUE TITLE
Error in AssetManager initialisation tracking

### ISSUE THREAD

**Issue by Svarr** [sll552/DiscordBee#134]
The [`MusicBeePlugin.DiscordTools.Assets.AssetManager.Init()`](https://github.com/sll552/DiscordBee/blob/c5fcb0f4a5d287dce1f7bcdb6e1079777ac3cac7/DiscordTools/Assets/AssetManager.cs#L24) method won't behave as intended in case of an error during `_assetUploader.Init()` or `_assetUploader.GetAssets()`. The line after the catch block will be executed in any case, overwriting the previous value with `true`.

I made sure and wrote a little test program:
```C#
using System;
class MainClass
{
  static void Main()
  {
    Console.WriteLine(Test());
  }

  static int Test()
  {
      int test = 0;
    try
    {
        ThrowStuff();
        test = -2;
    }
    catch
    {
        ++test;
    }
    return ++test;
  }

  static void ThrowStuff() => throw new NullReferenceException();
}
```
which compiles (if the line after the catch wouldn't be executed in case of a thrown exception, the compiler should complain about a missing return / not returning a value) and prints `2`. So both `++test` instructions get executed.

In conclusion, I suggest changing
https://github.com/sll552/DiscordBee/blob/c5fcb0f4a5d287dce1f7bcdb6e1079777ac3cac7/DiscordTools/Assets/AssetManager.cs#L26-L35
into
```C#
      try
      {
        await _assetUploader.Init();
        _ = _assetUploader.GetAssets();
        initialised = true;
      }
      catch
      {
        initialised = false;
      }
```

**Comment by sll552** [comment]
Thanks, nice catch, that's actually a good example for why you should not code too late at night :D
